### DISCUSSION TITLE
Can I create a custom executor?

### DISCUSSION THREAD

**Discussion by fa2k** [nextflow-io/nextflow#4022] [General]
I'm working on a SLURM cluster that requires `--mem-per-cpu`, and does not allow `--mem`.

Can I force nextflow to use a modified copy of SlurmExecutor.groovy (https://github.com/nextflow-io/nextflow/blob/master/modules/nextflow/src/main/groovy/nextflow/executor/SlurmExecutor.groovy) or otherwise override the SLURM executor?

I can of course set clusterOptions to set `--mem-per-cpu`, but many processes rely on the value of `memory`, so they won't work well if I set `memory` to `null`.

Or is there maybe an alternative solution? I'm also going to get in touch with the cluster admins and see if they allow to use `--mem` instead.

**Comment by bentsherman** [comment]
Hi @fa2k , there is actually a PR #4023 in the works to add a config option for `--mem-per-cpu`, so you won't have to create a whole new executor 😄

**Comment by fa2k** [comment]
That's fantastic! Thanks for the reply.
